discord/client.py

```diff
@@ -442,6 +442,30 @@ def _reconnect_ws(self):
     # login state management
 
     @asyncio.coroutine
+    def login_with_token(self, token):
+        self.token = token
+        if self.token is None:
+            raise LoginFailure('No Token Specified.')
+            yield from self.close()
+        else:
+            self.headers['authorization'] = token
+            self._is_logged_in.set()
```

Comment:
Unlike the other `login` method, this does not appear to do any sort of check that the token is actually valid. This seems like a mistake.